iree/test/e2e/regression/generate_e2e_matmul_tests.py

```diff
@@ -200,9 +230,58 @@ def get_test_generators(shapes_id: ShapesId):
                       acc=MatrixGenerator.RANDOM,
                       dynamicity=Dynamicity.STATIC),
     ]
+  if shapes_id == ShapesId.GPU_LARGE:
+    return [
+        TestGenerator(lhs=MatrixGenerator.IDENTITY,
+                      rhs=MatrixGenerator.IDENTITY,
+                      acc=MatrixGenerator.ZERO,
+                      dynamicity=Dynamicity.STATIC),
+        TestGenerator(lhs=MatrixGenerator.RANDOM,
+                      rhs=MatrixGenerator.RANDOM,
+                      acc=MatrixGenerator.RANDOM,
+                      dynamicity=Dynamicity.STATIC),
+    ]
   raise ValueError(shapes_id)
 
 
+@dataclasses.dataclass
+class TileWorkgroupSizePair:
+  tile_size: typing.List[int]
+  workgroup_size: typing.List[int]
+
+
+# Returns the list of CompilationInfo's to use for the CompilationInfoId.
+def get_test_compilation_infos(
+    compilation_info_id: CompilationInfoId) -> typing.List[CompilationInfo]:
+  if compilation_info_id == CompilationInfoId.LLVMGPUMatmulSimt:
+    tile_workgroup_size_pairs = [
+        TileWorkgroupSizePair([32, 128, 32], [32, 8, 1]),
+        TileWorkgroupSizePair([128, 64, 8], [16, 8, 1]),
+        TileWorkgroupSizePair([16, 256, 32], [64, 2, 1]),
+        TileWorkgroupSizePair([8, 32, 32], [8, 8, 1]),
+        TileWorkgroupSizePair([8, 128, 4], [32, 1, 1]),
+        TileWorkgroupSizePair([16, 64, 4], [16, 2, 1]),
+        TileWorkgroupSizePair([1, 128, 8], [32, 1, 1]),
```

Comment:
Actually this PR is pretty good in this respect! The CompilationInfo/ileWorkgroupSizePair info is explicitly mangled into the test function's name, see `generate_function_name`.  That is logged by the test runner. So, though it will require reading the logs, it will be clear what function failed, and it will be easy for someone familiar with this test suite, and feasible for someone without prior familiarity, to trace it back to the specific TileWorkgroupSizePair.

Reply:
> the generated CMakeLists would become enormous

If there are actually lots of test cases, that's okay. See for example this recently refactored test suite:
https://github.com/google/iree/tree/main/integrations/tensorflow/test/iree_tf_tests/math

The complexity I'm weary of here is using another python script to generate test cases, particularly now that this python script is 700 lines long. We already have multiple ways of defining and running tests (googletest, lit, check, etc.) split across two build systems, and this adds yet another. The TF integrations tests were using a cartesian product macro and it took quite a dedicated effort to move those over to the https://github.com/google/iree/blob/main/build_tools/testing/gen_test_matrix.py generator (https://github.com/google/iree/pull/6126) then to the https://github.com/google/iree/tree/main/integrations/tensorflow/test/python generator (https://github.com/google/iree/pull/8078). As we have a few of these, I'd like to see them look more alike and use the same underlying test runner (e.g. lit or check, depending on test configuration requirements).

Reply:
I don't want to stall this work too much, but we do have several efforts going on right now to simplify our infrastructure (mainly disconnecting compiler tests from the build phase and migrating from Kokoro to Buildkite), and I see signs here of us digging ourselves into another complexity hole.  I'd have to look more at the specifics of what these tests are trying to do to offer concrete suggestions, but I have a feeling that a combination of these could help:

* Move tests out of e2e/regression. They should not have been added there in the first place.
* Organize generator scripts across different test suites consistently
  * some combination of moving them to predictable places in the project structure and making them look and behave the same way
* Be okay with directories full of generated test files or verbose CMake files. Explicit tests are easy to understand at a glance
* Drop bazel support unless there is a really good reason to keep it

Reply:
good stuff! just to be clear, the things you point out here are preexisting, so this shouldn't block this PR now, right? The current state of these e2e matmul tests, including its location in `e2e/regression`, was introduced in PR #7347, reviewed by @benvanik and @GMNGeoffrey . We may need a meeting with all of us at the same time, to avoid continuing a telephone game.

Reply:
Most of it is preexisting, but this PR does increase iree/test/e2e/regression/generate_e2e_matmul_tests.py from 500 lines to 700 lines. That's the "step in the wrong direction" I'm worried about. There have been a few of these where I'm concerned about tech debt piling up. None of it feels like it should block any individual PR, but it all adds up and must be paid down at some point.

Reply:
Can you either file an issue or get us all in a meeting to record/discuss this?

Reply:
I think this discussion is beyond the scope of this PR. This PR only ads 14 tests to a set of existing tests (literally multiple hundreds) using existing build paths. In terms of overall scope, the complexity added here is minor compared to what already exists. 

Massive restructuring of tests should be handled separately.

Reply:
The new `CompilationInfo` argument and associated plumbing through this already complex test generation script is within the scope of this PR. It is adding more complexity and indirection to a component that is already too complex. That's going to take more work to refactor and clean up later and will increase the cost of test maintenance.

This change on its own seems okay to me now, but I'll keep asking for this e2e matmul testing to be simplified until that happens.